manifests/config/global.pp

```diff
@@ -18,15 +19,13 @@
   $flattendata = $newtype.keys.reduce({}) |$memo, $key| { $memo + {$key => $newtype[$key]["value"]} }
 
   unless empty($flattendata) {
-    concat::fragment { 'rsyslog::component::global_config::newtype':
-      order   => $::rsyslog::global_config_priority,
-      target  => "${::rsyslog::confdir}/${::rsyslog::target_file}",
-      content => epp('rsyslog/global_config.epp',
-      {
-        'type'   => 'rainerscript',
-        'config' => $flattendata,
-      }),
+    rsyslog::component::global_config { 'rainerscript':
+      *      => {
```

Comment:
I don't see the need for the `'*'` notation here - you are not creating the parameters dynamically, why not just list them normally?